src/charm.py

```diff
@@ -205,6 +207,110 @@ def _check_kf_profiles(self, interfaces):
 
         return kf_profiles
 
+    def main(self, event) -> None:
+        """Main entry point for the Charm."""
+        try:
+            self._check_container_connection()
+            self._check_model_name()
+            self._check_leader()
+            interfaces = self._get_interfaces()
+            kf_profiles = self._check_kf_profiles(interfaces)
+            self.handle_ingress(interfaces)
+        except CheckFailed as e:
+            self.model.unit.status = e.status
+            return
+        try:
+            self.unit.status = MaintenanceStatus("Creating k8s resources")
+            self.k8s_resource_handler.apply()
+        except ApiError:
+            self.logger.error(traceback.format_exc())
+            self.unit.status = BlockedStatus("kubernetes resource creation failed")
+            return
+        self.handle_ingress(interfaces)
+        kf_profiles = list(kf_profiles.get_data().values())[0]
+        self.profiles_service = kf_profiles["service-name"]
+        try:
+            self._update_layer()
+        except CheckFailed as e:
+            self.model.unit.status = e.status
+            return
+        self.model.unit.status = ActiveStatus()
+
+    def _get_sidebar_config(self) -> Dict:
+        try:
+            current_configmap = self.k8s_resource_handler.lightkube_client.get(
+                ConfigMap, name=self._context["configmap_name"]
+            )
+            old_sidebar_config = json.loads(current_configmap.data["links"])
+        except ApiError as e:
+            self.logger.warning(
+                f"PROBLEM during configmap retrieval {e}. USING BASE CONFIG"
+            )
+            old_sidebar_config = json.loads(BASE_SIDEBAR)
+        return old_sidebar_config
+
+    def _update_sidebar_config(self, links_data: Dict) -> None:
+        self._context["links"] = json.dumps(links_data)
+        try:
+            self.k8s_resource_handler.context = self._context
+            self.k8s_resource_handler.template_files = [
+                self._resource_files["config_maps"]
+            ]
+            self.k8s_resource_handler.apply()
+        except ApiError as e:
+            self.unit.status = BlockedStatus(
+                f"kubernetes resource creation failed with {e}"
+            )
+
```

Comment:
Didn't have a good place to put this... We should also update the sidebar whenever `main` is called.  Some of the events that can trigger `main`, for example `juju config kubeflow-dashboard dashboard-configmap=something-new` affect the sidebar and its contents.

Also I didn't see a place that would handle the above config change (where we need to delete the old configmap and create a new one with the new name), but I might have missed it

Reply:
Resolved Thank you for pointing out.